src/main/kotlin/org/sol4k/VersionedTransaction.kt

```diff
@@ -0,0 +1,81 @@
+package org.sol4k
+
+import org.sol4k.Constants.SIGNATURE_LENGTH
+import org.sol4k.exception.SerializationException
+import java.io.ByteArrayOutputStream
+import java.math.BigDecimal
+import java.math.RoundingMode
+import java.util.Base64
+import kotlin.math.max
+
+class VersionedTransaction(
+    val message: TransactionMessage,
+    val signatures: MutableList<String>,
+) {
+
+    fun sign(keypair: Keypair) {
+        val data = message.serialize()
+        val signature = keypair.sign(data)
+        for (i in 0 until message.header.numRequireSignatures) {
+            val a = message.accounts[i]
```

Comment:
hello @Tougee , sorry if that's stupid question but, let's say we have only 1 signer and we set `numRequireSignatures=1` this will check account[0] and account[1] only. I'm a bit confused here. Can you please clarify this for me? 

Thanks

Reply:
thanks for your review @tufanbarisyildirim, this will only check account[0] if `numRequireSignatures=1`.

Reply:
So this means, you have to put signer account to the top of the list if you have only 1 signer?

Reply:
yes, signer should be on the top.